aesara/tensor/random/basic.py

```diff
@@ -81,6 +81,19 @@ def __call__(self, loc=0.0, scale=1.0, size=None, **kwargs):
     def rng_fn(cls, rng, loc, scale, size):
         return stats.halfnorm.rvs(loc, scale, random_state=rng, size=size)
 
+    @classmethod
+    def numba_fn(cls, np_random_fn_name, np_input_names, np_global_env):
+        np_global_env["abs_val"] = abs
+        np_global_env["standard_norm"] = np.random.standard_normal
+
+        src = f"""
+@numba_vectorize
+def {np_random_fn_name}({np_input_names}):
+    return abs_val(standard_norm({np_input_names}))
+"""
```

Comment:
This is something I tried (without much success) as an attempt to support custom implemented distributions. Somehow the output from this function gets classified as a `pyobject` rather than the `numba` type of the output value. (Even though both of the functions used inside are supported by Numba individually.)

Note: This particular implementation is taken from `scipy.stats` which does something pretty close to this.

https://github.com/scipy/scipy/blob/36af73c98f71f40bc69efbeb4501b1f2b9c3d00c/scipy/stats/_continuous_distns.py#L3504